### ISSUE TITLE
Add kabyle language

### ISSUE THREAD

**Issue by ghost** [ZJaume/heli-otr#11]
Hello,

Is it possible to add Kabyle `kab` language or should I [train a model](https://github.com/ZJaume/heliport/blob/main/docs/train.md) ? I can provide help. Kabyle language uses latin alphabet with some extra chars. OpenLID uses `kab_Latn`.

I git cloned the repo to see if I can add it.

Thank you !

**Comment by ghost** [comment]
I'm working on it.

**Comment by ghost** [comment]
PR https://github.com/ZJaume/heliport/pull/12

**Comment by ghost** [comment]
Test and screenshot here : https://mstdn.fr/@ButterflyOfFire/115275745527393751

**Comment by ZJaume** [comment]
Hi @BoFFire, thank you very much for your addition!

The main goal of this project is to just be a re-implementation of HeLI-OTS, as I don't have the time that is required to carefully curate and expand the set of supported languages. However, I'm willing to accept new language to be officially supported if they can be documented and achieve performance similar to the latest additions (close to 0.99 F1 on a test set like FLoRES). As noted in the training guide:
> When choosing the language or set of languages to add, it is recommended to look for similar languages that may already have support by the tool or may not be included in the set. In both cases, all the languages that are similar should be taken into account and diagnose performance on each one. As similar languages are the more probable ones to cause alterations in the tool accuracy. A good methodology for adding languages can be found in [Jauhiainen et. al.](https://aclanthology.org/2024.sigul-1.15/).

Regarding similar languages, I think there are no other Berber languages supported by heliport, and according to Wikipedia they form a dialect continuum, so adding support to only one of them we might have the risk of all or some other Berber languages without support to be classified as Kabyle. So it would be nice if this could be mitigated and/or documented. If you have further knowledge about this topic that can help decide a strategy on this, it would be great. For example, OpenLID has data for another Berber language written in Latin like Tamasheq and a couple more using Tifinagh (Tamasheq also and Standard Moroccan Amazigh).

**Comment by ghost** [comment]
Hi @ZJaume Thank you for your answer and I understand your concerns.

About FLORES dataset, I'm leading the team behind reviewing the FLORES dataset for OLDI project. The work is not yet done but is in progress.

Wikipedia has a Wikipedia in kabyle since 2008 under the domain kab.wikipedia.org

We use OPENLID v2 along with GlotLID determine if a sentence is in kabyle language. They use `kab_Latn`, we simplify it as `kab`.

Kabyle is written in Latin alphabet and we contribute to CLDR under `kab` locale. No tifinagh script.

GlotLID supports some other Berber languages in fact. And I'll be happy to follow your recommendations.

Pinging @MedAymenF for advice.

**Comment by MedAymenF** [comment]
@BoFFire To avoid misclassifying other Amazigh languages and dialects as Kabyle, you can add data for the major ones found online, roughly ordered by similarity to Kabyle: Shawiya, Central Atlas Tamazight, Tarifit, Tachelhit, and Tamasheq.

Most of them are on Tatoeba or have Wikipedias (or still incubated). Central Atlas Tamazight, while being arguably the closest Moroccan dialect to Kabyle, has very little data written in Latin. However, I can provide you with some data sources if you're interested in adding it.

**Comment by ghost** [comment]
Thank you @MedAymenF :) Yes, seems wise. I'm interested. That's what I asked our team today, at least for Chaoui : https://mstdn.fr/@ButterflyOfFire/115292616298132360

**Comment by MedAymenF** [comment]
Great! let's use a faster way of communication to prepare the data, then we can come back to this Issue/PR. I sent you a DM on Discord. Do you prefer Telegram or WhatsApp?

**Comment by ZJaume** [comment]
@MedAymenF thank you very much for your help!

I want point out some datasets that both of you may be already aware of, or may be already included in GlotLID or OpenLID, but just in case.
- Several langs:
  - https://opus.nlpl.eu/bible-uedin/corpus/version/bible-uedin
  - https://huggingface.co/TutlaytAI/datasets
  - https://huggingface.co/datasets/mozilla-foundation/common_voice_17_0
- Tamashek:
  - https://huggingface.co/datasets/michsethowusu/tamasheq-emotions-corpus
- Kabyle:
  - https://huggingface.co/datasets/TutlaytAI/kabyle_asr

Training data does not need to be restricted to Tatoeba and Wikipedia, it can be text from any other ML task as long as the data has not been automatically collected and has risks of other languages contamination (web crawled e.g.). And while heliport can have models that are trained on a few thousand sentences, it is recommended to have several tens of thousands or more. **Cleaning** prior to training is also advised (e.g. running the heliport on the training data before adding any Berber language to see if there is contamination of other languages, like English, or sentences that do not contain textual information).

> About FLORES dataset, I'm leading the team behind reviewing the FLORES dataset for OLDI project. The work is not yet done but is in progress.

If there are issues in the FLORES dataset related to translation itself, it's fine to use it as a test set for langid.

**Comment by MedAymenF** [comment]
Thank you for the suggestions @ZJaume. Many of the datasets you've listed are actually just derivatives of Tatoeba, OLDI SEED and FLORES datasets, which continue to be the largest clean corpora for many of these languages.

When you say "it is recommended to have several tens of thousands or more", you mean _per language_ right?

**Comment by ZJaume** [comment]
> When you say "it is recommended to have several tens of thousands or more", you mean per language right?

Yes.

Edit: but the more the better.

**Comment by ghost** [comment]
- OK, we gathered some rare resources never seen elsewhere in kabyle language and never collected, published on our community datasets : https://huggingface.co/Imsidag-community/datasets

- I'm working on better cleaner/normalizer and standardizer for kabyle language to meet the CLDR/Unicode alphabet set.
- Waiting for datasets about other active berber languages, unfortunately I have no idea where to find not only clean but quite a decent amount of resources.

**Comment by ghost** [comment]
May be I should close this for now and try later.

**Comment by ZJaume** [comment]
May I ask for the full list of languages that you have so far and the training size for each one? (number of sentences and words)
